core/src/nucleus/ribosome/api/get_entry.rs

```diff
@@ -9,17 +10,21 @@ use std::sync::mpsc::channel;
 use wasmi::{RuntimeArgs, RuntimeValue, Trap};
 
 #[derive(Deserialize, Default, Debug, Serialize)]
-struct GetArgs {
-    key: String,
+struct GetAppEntryArgs {
+    key: HashString,
 }
 
+/// ZomeApiFunction::GetAppEntry function code
+/// args: [0] encoded MemoryAllocation as u32
+/// Expected complex argument: GetEntryArgs
+/// Returns an HcApiReturnCode as I32
 pub fn invoke_get_entry(
     runtime: &mut Runtime,
     args: &RuntimeArgs,
 ) -> Result<Option<RuntimeValue>, Trap> {
     // deserialize args
     let args_str = runtime_args_to_utf8(&runtime, &args);
-    let res_entry: Result<GetArgs, _> = serde_json::from_str(&args_str);
+    let res_entry: Result<GetAppEntryArgs, _> = serde_json::from_str(&args_str);
     // Exit on error
     if res_entry.is_err() {
```

Comment:
I saw this earlier in scrum, I was just wondering if there was any reason we are using this pattern of error handling instead of match?

Reply:
I find code a lot clearer and easier to debug if you exit early when the pre-conditions are not met.
The problem with doing a match early in a function is that the rest of the function code will be within the match arm and thus indented. If you keep doing that you end up with 'pyramids' of indented code, which gets really hard to read.

Reply:
We could do a `match` that does not nest code by doing it in the `let`, but I find this a bit more complicated to understand when reading it. Matter of taste I guess.